tests/test_attachment.py

```diff
@@ -0,0 +1,130 @@
+    b"\x8eR\x94\xad\x91+\xf1f@\x81\x9a\xcc{\x89\x84;\xd7\xdf\xffk\xea\xc0\xa8\xb9\xe1\xda\xea"
+    b"\xe0\x07\x83\x12&\r\xae)$\x1b\xdc\xe9\xe5\xf0\xb0\xe0a\x0bP\xf1\x04iw\x03a\x86\rM\xe8"
+    b"\xe9\xe0i\x08\xb1\xa0\x00\x00\x00"
+)
+
+
+async def test_save_text(doc):
+    await doc.save()
+    att = doc.attachment("lipsum.txt")
+    await att.save(text, "text/plain")
+
+
+async def test_save_binary(doc):
+    await doc.save()
+    att = doc.attachment("image.webp")
+    await att.save(image, "image/webp")
+
+
+async def test_save_on_unfetched(doc):
+    await doc.save()
+
+    new_doc = Document(doc._database, doc["_id"])
+
+    att = new_doc.attachment("lipsum.txt")
+    with pytest.raises(ValueError):
+        await att.save(text, "text/plain")
+
+    new_doc = await doc._database.get(doc["_id"])
+    att = new_doc.attachment("image.webp")
+    await att.save(text, "text/plain")
+
+
+async def test_get(doc):
+    await doc.save()
+    await doc.attachment("image.webp").save(image, "image/webp")
+
+    att = doc.attachment("image.webp")
+    data = await att.fetch()
+    assert data == image
+    assert att.content_type == "image/webp"
+
+
+async def test_get_on_unfetched(doc):
+    await doc.save()
+    await doc.attachment("image.webp").save(image, "image/webp")
+
+    new_doc = Document(doc._database, doc["_id"])
+    att = new_doc.attachment("image.webp")
+    data = await att.fetch()
+    assert data == image
+
+
+async def test_update(doc):
+    await doc.save()
+
+    doc["value"] = 42
+    await doc.save()
+
+    att = doc.attachment("lipsum.txt")
+    await att.save(text, "text/plain")
+
+    att = doc.attachment("image.webp")
+    await att.save(image, "image/webp")
+
+    doc["value"] = 43
+    await doc.save()
+
+    new_doc = await doc._database.get(doc["_id"])
+    assert new_doc["value"] == 43
+    assert await new_doc.attachment("lipsum.txt").fetch() == text
+    assert await new_doc.attachment("image.webp").fetch() == image
+
+
+async def test_conflict(doc):
+    await doc.save()
+
+    outdated = await doc._database.get(doc["_id"])
+
+    doc["value"] = 42
+    await doc.save()
+
+    with pytest.raises(ConflictError):
+        await outdated.attachment("image.webp").save(image, "image/webp")
+
+    await doc.attachment("image.webp").save(image, "image/webp")
+
+
+async def test_conflict_on_update(doc):
+    await doc.save()
+
+    outdated = await doc._database.get(doc["_id"])
+
+    att = doc.attachment("lipsum.txt")
+    await att.save(text, "text/plain")
+
+    with pytest.raises(ConflictError):
+        await outdated.attachment("image.webp").save(image, "image/webp")
+
+    await doc.attachment("image.webp").save(image, "image/webp")
+
+
+async def test_delete(doc):
+    await doc.save()
+    att = doc.attachment("lipsum.txt")
+    await att.save(text, "text/plain")
+
+    await att.delete()
+
+    att = doc.attachment("lipsum.txt")
+    with pytest.raises(NotFoundError):
```

Comment:
I'd also `assert not await att.exists()`

Reply:
Good idea.